internal/services/firewall/firewall_policy_rule_collection_group_resource.go

```diff
@@ -757,7 +758,13 @@ func flattenFirewallPolicyRuleCollection(input *[]firewallpolicyrulecollectiongr
 
 			var action string
 			if rule.Action != nil {
-				action = string(pointer.From(rule.Action.Type))
+				// todo 4.0 change this from DNAT to Dnat
+				// doing this because we hardcode Dnat for https://github.com/Azure/azure-rest-api-specs/issues/9986
+				if strings.EqualFold(string(pointer.From(rule.Action.Type)), "Dnat") {
+					action = "Dnat"
+				} else {
+					action = string(pointer.From(rule.Action.Type))
+				}
```

Comment:
since Firewall is using the new base layer, this shouldn't be needed - the base layer should be updating Action to be `DNAT` as defined in the Swagger - what's the issue this is fixing?

Reply:
The issue is that we're specifying `Dnat` in the config files which should mean our users are too 

<img width="252" alt="image" src="https://github.com/hashicorp/terraform-provider-azurerm/assets/6515518/1f38e49a-d015-41ab-bebb-a06f340b2ac6">